src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -961,7 +958,8 @@ bool emberAfOperationalCredentialsClusterAttestationRequestCallback(app::Command
     attestationElementsSpan = MutableByteSpan{ attestationElements.Get(), attestationElementsLen };
     err = Credentials::ConstructAttestationElements(certDeclSpan, attestationNonce, timestamp, kEmptyFirmwareInfo,
                                                     emptyVendorReserved, attestationElementsSpan);
-    VerifyOrExit((err == CHIP_NO_ERROR) && (attestationElementsSpan.size() <= kMaxRspLen), finalStatus = Status::Failure);
+    VerifyOrExit((err == CHIP_NO_ERROR) && (attestationElementsSpan.size() <= Credentials::kMaxRspLen),
```

Comment:
This is part of the contract of ConstructAttestationElements: it is not supposed to make the span slarger than kMaxRspLen. The check should be in ConstructAttestationElements

Reply:
moved this check into DeviceAttestationConstructor.cpp